### DISCUSSION TITLE
How do I display an XYChart without empty columns

### DISCUSSION THREAD

**Discussion by AlexHomebrew** [jfree/jfreechart#376] [Q&A]
I an using XYChart to display the current state of a bunch of sensors, where the X-Axis is the number of seconds since the start.
![SensorGraph1](https://github.com/jfree/jfreechart/assets/6464337/bff72372-edb2-48f6-80d6-e1edf8844a4b)
The problem is there occasionally is a long period between 2 data columns (XYSeries) as in the image between 42 and 49. It can be much longer which wastes horizontal space. I would like to not display the tick marks for columns that have no XYSeries, so the chart would look like
![SensorGraph2](https://github.com/jfree/jfreechart/assets/6464337/3be84930-77e3-4f40-88eb-ab9529b8fc31)
I have looked at the Axis objects, the Tick objects, and various other objects but can't figure out how to do this.
Should I change to a ScatterPlot and connect the dots?
Thanks
Alex

**Comment by trashgod** [comment]
I try to avoid eliding data in favor of controls that let the user control focus, as suggested [here](https://stackoverflow.com/a/44151032/230513), [here](https://stackoverflow.com/a/41544007/230513) and [here](https://stackoverflow.com/a/6849654/230513). Having said that, you may also want to look at a custom [`Timeline`](https://www.jfree.org/jfreechart/api/javadoc/org/jfree/chart/axis/Timeline.html), outlined [here](https://stackoverflow.com/a/55312264/230513).

**Reply by AlexHomebrew** [reply]
@trashgod The pan and zoom functions are great, but I'm not really eliding data, there is no data between seconds 42 and 49. I will look at a custom TimeLine, thanks.
